tests/test_qnode.py

```diff
@@ -524,6 +525,31 @@ def circuit():
 class TestTapeConstruction:
     """Tests for the tape construction"""
 
+    def test_shots_override_warning(self):
+        """Test that a warning is raised when both set_shots transform and shots parameter are used."""
+        dev = qml.device("default.qubit", wires=1)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.Hadamard(wires=0)
+            return qml.sample(qml.PauliZ(0))
+
+        # First apply set_shots to override device shots
+        modified_circuit = qml.set_shots(circuit, shots=50)
+        assert modified_circuit._shots == qml.measurements.Shots(50)
+        assert modified_circuit._shots_override_device is True
+
+        # Then try to pass shots parameter when calling the QNode
+        with pytest.warns(
+            UserWarning,
+            match="Both 'shots=' parameter and 'set_shots' transform are specified. "
+            "The transform will take precedence over 'shots=25'.",
+        ):
+            result = modified_circuit(shots=25)
+
+        # Verify that the set_shots value (50) was used, not the parameter value (25)
+        assert len(result) == 50
```

Comment:
Mind moving this test down into the `TestSetShots` class?

Reply:
https://github.com/PennyLaneAI/pennylane/pull/7627/commits/b1d7cf0d577a4def95544b8fdf5c36348ad03586